src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -32,15 +32,32 @@ import Foundation
 struct KeyValueViewModel {
     let key: String
     let value: String
+    let valueNilable: Bool
 
-    init(from param: Parameter, with model: CodeModel) {
-        self.key = param.serializedName!
-        if let constant = param.schema as? ConstantSchema {
-            self.value = "\"\(constant.value.value)\""
+    init(from param: Parameter, with model: CodeModel, and operation: Operation, using key: String? = nil) {
+        self.key = key ?? param.serializedName!
+
+        if let constantSchema = param.schema as? ConstantSchema {
+            let isString: Bool = constantSchema.valueType.type == AllSchemaTypes.string
+            let val: String = constantSchema.value.value
+
+            self.value = isString ? "\"\(val)\"" : "\(val)"
+            self.valueNilable = false
+        } else if let signatureParameter = operation.signatureParameters(for: param.name) {
+            self.value = param.name
```

Comment:
Does this not need the same checking as other to determine whether it is a string or not?